### PR TITLE
Forbid Generic variables.

### PR DESCRIPTION
**Pull request by dekimir**

### CODE REVIEW

**Comment by dekimir** [comment]
@umar456 PTAL

**Comment by umar456** [comment]
Looks good.

On a side note. I wanted to talk about how to represent the instructions names in error messages. I have been representing them by removing the Op prefix. What are your thoughts?

**Comment by dekimir** [comment]
I don't have a strong opinion either way.  I've been using the Op prefix because the spec always uses it when setting the rules.  Some of my error messages are copied verbatim from the spec.

**Comment by umar456** [comment]
Makes sense.

File: test/Validate.Storage.cpp [hunk 1]
```diff
@@ -141,4 +141,34 @@ INSTANTIATE_TEST_CASE_P(MatrixOp, ValidateStorage,
...
+TEST_F(ValidateStorage, GenericVariableOutsideFunction) {
+  const auto str = R"(
+          OpCapability Kernel
+          OpMemoryModel Logical OpenCL
+%intt   = OpTypeInt 32 1
+%ptrt   = OpTypePointer Function %intt
+%var    = OpVariable %ptrt Generic
+)";
+  CompileSuccessfully(str);
+  ASSERT_EQ(SPV_ERROR_INVALID_LAYOUT, ValidateInstructions());
```
Comment (umar456) [inline test/Validate.Storage.cpp]: I don't think this should be a layout error

Comment (dekimir) [inline test/Validate.Storage.cpp]: An instruction error, perhaps?

Comment (umar456) [inline test/Validate.Storage.cpp]: Sounds good to me.

Comment (dneto0) [inline test/Validate.Storage.cpp]: Or the catchall INVALID_BINARY could work too.  Unless we have something more specific that is still accurate.

**Comment by dneto0** [comment]
LGTM

**Comment by dekimir** [comment]
Changed to `INVALID_BINARY`.

**Pull request merged**